### ISSUE TITLE
private collection with copied charters can create cardinality error

### ISSUE THREAD

**Issue by GVogeler** [icaruseu/mom-ca#750]
A user collection (my-collection) with copied charters can created an cardinality error:
```
cardinality error: sequence with more than one item is not allowed here [at line 161, column 26, source: /db/XRX.live/mom/app/collection/mycharters.xqm]
In function:
	mycharters:entries(item()*)
```
e.g.: http://monasterium.net/mom/7d89914c-c737-4852-8654-cb2bda5c466f/my-collection

**Comment by GVogeler** [comment]
As https://github.com/icaruseu/mom-ca/blob/master/my/XRX/src/mom/app/collection/mycharters.xqm#L38-L49 includes sorting by date, I would assume, that the private charter creates via EditMOM3 `cei:date` and `cei:dateRange` which would be evaluated by https://github.com/icaruseu/mom-ca/blob/master/my/XRX/src/mom/app/collection/mycharters.xqm#L45 with a resultset of two, which can not be evaluated by the order clause.
This assumption is supported by an experiment with the XQuery in the function with the collection entioned above: executed as stand alone it works if an index `[1]` is added to
`let $date := $cei-text//cei:issued/(cei:date/@value|cei:dateRange/@from)`.

**Comment by GVogeler** [comment]
The most complete solution seems to be to use the solution from https://github.com/icaruseu/mom-ca/issues/679#issuecomment-348707130

**Comment by GVogeler** [comment]
Btw: the reuse of the date/dateRange-selection would be an argument to move it into a function e.g. in then data.xqm module.

**Comment by GVogeler** [comment]
solved by #751
